include/avk/commands.hpp

```diff
@@ -1113,23 +1113,34 @@ namespace avk
 	{
 		avk::resource_argument<avk::semaphore_t> mWaitSemaphore;
 		avk::stage::pipeline_stage_flags mDstStage;
+		uint64_t mValue;
+
+		semaphore_wait_info& at_value(uint64_t aValue) {
+			mValue = aValue;
+			return *this;
+		}
 	};
 
 	inline semaphore_wait_info operator>> (avk::resource_argument<avk::semaphore_t> a, avk::stage::pipeline_stage_flags b)
 	{
-		return semaphore_wait_info{ std::move(a), b };
+		return semaphore_wait_info{ std::move(a), b, 0 };
 	}
 
-
 	struct semaphore_signal_info
 	{
 		avk::stage::pipeline_stage_flags mSrcStage;
 		avk::resource_argument<avk::semaphore_t> mSignalSemaphore;
+		uint64_t mValue;
+
+		semaphore_signal_info& to_value(uint64_t aValue) {
+			mValue = aValue;
+			return *this;
+		}

```

Comment:
...same here, but I have to agree that the `at_value`/`to_value` is not easy to get into `class semaphore_t`.

So, I guess we should rather go back to `operator|` or use `operator>>` once again (as in the source code that I'm about to push).

Reply:
As mentioned above I now use `operator,` to create `semaphore_value_info` in a concise way, which can then be used together with `operator>>` to create `semaphore_wait_info` and `semaphore_signal_info`.